frontend/src/components/Subscription.js

```diff
@@ -1,53 +1,88 @@
+                }));
+                setSubscriptions(parsedSubscriptions);
+            })
+            .catch(error => {
                 console.error('Error fetching subscriptions:', error);
-            } finally {
+            })
+            .finally(() => {
                 setIsLoading(false);
-            }
-        };
+            });
 
-        fetchSubscriptions();
+            fetch('http://localhost:5000/api/subscription/apps')
+            .then(response => response.json())
+            .then(appsData => setApplications(appsData))
+            .catch(error => console.error('Error fetching apps:', error));
     }, []);
 
+    const getAppName = (appId) => {
+        const app = applications.find(app => app.appId === appId);
+        return app ? app.appName : 'Unknown';
+    };
+
     if (isLoading) {
         return <div>Loading subscriptions...</div>;
     }
 
     return (
         <div className="subscription-table">
             <h2>Your Subscriptions</h2>
+
+            <button onClick={handleAddSubscription} className="btn btn-primary">
+                Add New Subscription
+            </button>
+
             <table>
                 <thead>
                     <tr>
-                        <th>App Name</th>
+                        <th>App ID</th>
                         <th>Cost</th>
                         <th>Subscription Date</th>
                         <th>Renewal Date</th>
-                        <th>Reminder Date</th>
+                        <th>Actions</th>
                     </tr>
                 </thead>
                 <tbody>
                     {subscriptions.map((subscription) => (
-                        <tr key={subscription.id}>
-                            <td>{subscription.appName}</td>
-                            <td>{subscription.cost}</td>
-                            <td>{new Date(subscription.subscriptionDate).toLocaleDateString()}</td>
-                            <td>{new Date(subscription.renewalDate).toLocaleDateString()}</td>
-                            <td>{new Date(subscription.reminderDate).toLocaleDateString()}</td>
+                        <tr key={subscription.subscriptionId}>
+                            <td data-label="App Name">{getAppName(subscription.appId)}</td>
+                            <td data-label="Cost">{subscription.cost}</td>
+                            <td data-label="Subscription Date">{new Date(subscription.subscriptionDate).toLocaleDateString()}</td>
+                            <td data-label="Renewal Date">{new Date(subscription.renewalDate).toLocaleDateString()}</td>
+                            <td data-label="Actions">
+                                <button onClick={() => handleEditSubscription(subscription)} className="btn btn-secondary">Edit</button> {/* Edit button */}
+                            </td>
                         </tr>
                     ))}
                 </tbody>
```

Comment:
We can create cards instead of table to give a better look to the subscription page. Lets add that point in the design modifications.

Reply:
Noted..